Scripts/Pointer/StraightPointer.cs

```diff
@@ -0,0 +1,176 @@
+            {
+                return true;
+            }
+
+            if (tracerVisibility == VisibilityType.AlwaysOff && cursorVisibility == VisibilityType.AlwaysOff)
+            {
+                return false;
+            }
+
+            return Active;
+        }
+
+        /// <summary>
+        /// The Deactivate method turns off the pointer.
+        /// </summary>
+        public override void Deactivate()
+        {
+            base.Deactivate();
+            SetElementState();
+        }
+
+        protected virtual void OnEnable()
+        {
+            Active = activateOnEnable;
+            if (Active)
+            {
+                Activate();
+            }
+            SetElementState();
+        }
+
+        protected virtual void OnDisable()
+        {
+            SetElementState();
+        }
+
+        protected virtual void Update()
+        {
+            if (IsVisible())
+            {
+                float tracerLength = CalculateTracerLength();
+                FormatElements(tracerLength);
+                CheckValidity();
+            }
+            SetElementState();
+            CursorData = (validCursor.transform != null ? validCursor.transform : null);
+        }
+
+        /// <summary>
+        /// The SetElementState method determines the active state of the pointer elements.
+        /// </summary>
+        protected virtual void SetElementState()
+        {
+            SetElementVisible(tracerVisibility, validTracer, invalidTracer);
+            SetElementVisible(cursorVisibility, validCursor, invalidCursor);
+        }
+
+        /// <summary>
+        /// The SetElementVisible method determines the visibility state of the given element.
+        /// </summary>
+        /// <param name="elementType">The VisibilityType parameter of the element to determine visibility for.</param>
+        /// <param name="validElement">The GameObject to set active is the element should be valid.</param>
+        /// <param name="invalidElement">The GameObject to set active is the element should be invalid.</param>
+        protected virtual void SetElementVisible(VisibilityType elementType, GameObject validElement, GameObject invalidElement)
+        {
+            if (IsElementVisible(elementType))
+            {
+                SetElementActive(validElement, Valid);
+                SetElementActive(invalidElement, !Valid);
+            }
+            else
+            {
+                SetElementActive(validElement, false);
+                SetElementActive(invalidElement, false);
+            }
+        }
+
+        /// <summary>
+        /// The SetElementActive method determines the active state of the given element.
+        /// </summary>
+        /// <param name="element">The GameObject of which to set it's active state.</param>
+        /// <param name="state">The state to set on the GameObject.</param>
+        protected virtual void SetElementActive(GameObject element, bool state)

```

Comment:
Maybe this should be an extension method on any `GameObject` instead?

Reply:
I don't think this will work as an extension method because it's checking if the gameobject is null

if the gameobject was null then you wouldn't have extension methods available on it would you? or does an extension method on a gameobject still allow for the object calling the extension to be null? if so then yeah i agree extension is better

Reply:
Yup that works with extension methods and is actually something one has to be aware of when writing extension methods.

Basically all that needs to check out is the type as it's compile-time, the value isn't checked.